FreeRTOS-Plus/Source/FreeRTOS-Plus-CLI/FreeRTOS_CLI.c

```diff
@@ -282,41 +284,62 @@ const char *pcReturn = NULL;
 
 static BaseType_t prvHelpCommand( char *pcWriteBuffer, size_t xWriteBufferLen, const char *pcCommandString )
 {
-static const CLI_Definition_List_Item_t * pxCommand = NULL;
-BaseType_t xReturn;
-
-	( void ) pcCommandString;
-
-	if( pxCommand == NULL )
-	{
-		/* Reset the pxCommand pointer back to the start of the list. */
-		pxCommand = &xRegisteredCommands;
-	}
-
-	/* Return the next command help string, before moving the pointer on to
-	the next command in the list. */
-	strncpy( pcWriteBuffer, pxCommand->pxCommandLineDefinition->pcHelpString, xWriteBufferLen );
-	pxCommand = pxCommand->pxNext;
-
-	if( pxCommand == NULL )
-	{
-		/* There are no more commands in the list, so there will be no more
-		strings to return after this one and pdFALSE should be returned. */
-		xReturn = pdFALSE;
-	}
-	else
-	{
-		xReturn = pdTRUE;
-	}
-
+    
+    BaseType_t xReturn = pdFALSE;
+    static const CLI_Definition_List_Item_t *pxCommand = NULL;
+    
+    BaseType_t lParameterStringLength;
+    char * pcParameter = (char *)FreeRTOS_CLIGetParameter ( pcCommandString, 1, &lParameterStringLength );
+
+    if (pcParameter != NULL)
+    {
+        pcParameter[lParameterStringLength] = '\0';
+        for( pxCommand = &xRegisteredCommands; pxCommand != NULL; pxCommand = pxCommand->pxNext )
+		{
+            if( strncmp( pcParameter, pxCommand->pxCommandLineDefinition->pcCommand, lParameterStringLength-1 ) == 0 )
+            {
+                strncpy( pcWriteBuffer, pxCommand->pxCommandLineDefinition->pcHelpString, xWriteBufferLen );
+                strncpy( pcWriteBuffer + strlen(pcWriteBuffer), "\r\n\n", sizeof ("\r\n\n") );
+                return pdFALSE;
+            }
+        }
+        strncpy( pcWriteBuffer, "Command not recognised.  Enter 'help' to view a list of available commands.\r\n\r\n", xWriteBufferLen );
+    }
+    else
+    {
+        if( pxCommand == NULL )
+        {
+            /* Reset the pxCommand pointer back to the start of the list. */
+            pxCommand = &xRegisteredCommands;
+        }
+
+        /* Return the next command help string, before moving the pointer on to
+        the next command in the list. */
+
+        strncpy( pcWriteBuffer, pxCommand->pxCommandLineDefinition->pcDescriptionString, xWriteBufferLen );

```

Comment:
From what I understand, here it should be pcHelpString to indicate the short description? Because this is the "help" case.